src/adapter/src/explain/optimizer_trace.rs

```diff
@@ -133,24 +138,62 @@ impl OptimizerTrace {
                     .collect();
                 rows
             }
-            Some(path) => {
+            ExplainStage::PlanInsights => {
+                if format != ExplainFormat::Json {
+                    coord_bail!("EXPLAIN PLAN INSIGHTS only supports JSON format");
+                }
+
+                let mut text_traces = collect_all(ExplainFormat::Text)?;
+                let mut json_traces = collect_all(ExplainFormat::Json)?;
+                let global_plan = self.collect_global_plan();
+                let fast_path_plan = self.collect_fast_path_plan();
+
+                let mut get_plan = |name: NamedPlan| {
+                    let text_plan = match text_traces.remove(name.path()) {
+                        None => "<unknown>".into(),
+                        Some(entry) => entry.plan,
+                    };
+                    let json_plan = match json_traces.remove(name.path()) {
+                        None => serde_json::Value::Null,
+                        Some(entry) => serde_json::from_str(&entry.plan).map_err(|e| {
+                            AdapterError::Unstructured(anyhow::anyhow!("internal error: {e}"))
+                        })?,
+                    };
+                    Ok::<_, AdapterError>(serde_json::json!({
+                        "text": text_plan,
+                        "json": json_plan,
+                    }))
+                };
+
+                let output = serde_json::json!({
+                    "plans": {
+                        "raw": get_plan(NamedPlan::Raw)?,
+                        "optimized": {
+                            "global": get_plan(NamedPlan::Global)?,
```

Comment:
Are we not interested in the `optimize/fast_path` result here as well? A lot of `SELECT` queries are executed through that path and per default `EXPLAIN <SELECT>` would render the plan located at `optimize/fast_path` instead of `optimize/global`.

Reply:
Ah, that's a good idea. @mjibson do you mind stitching this in as `plans.optimized.fast_path`?

Reply:
Done.